jetty-ee10/jetty-ee10-annotations/src/main/java/org/eclipse/jetty/ee10/annotations/AnnotationParser.java

```diff
@@ -520,6 +531,44 @@ public FieldVisitor visitField(final int access,
         }
     }
 
+    /**
+     * Check whether the classname is located inside its containingResource at
+     * the location that matches its package. The comparison accounts for classes
+     * located within WEB-INF/classes.
+     * @param containingResource the Resource that contains the class
+     * @param classResource the Resource representing the class
+     * @param classname the package-qualified name of the class
+     * @return true if the containingResource contains a class file at the location matching the package name of the class
+     */
+    public static boolean checkClassContainment(Resource containingResource, Resource classResource, String classname)
+    {
+        if (containingResource == null || classResource == null)
+            return false;
+        Path relative = containingResource.getPathTo(classResource);
+        if (relative == null)
+            return false; //unable to be verified
+
+        StringTokenizer tokenizer = new StringTokenizer(classname, "/", false);
```

Comment:
Doesn't the classname have '.' separators rather than '/'?

If it has '/' separators, then maybe we should split it with Path?

Reply:
The classname when referenced via a resource has slashes.